unified-scheduler-pool/src/lib.rs

```diff
@@ -166,6 +171,25 @@ where
                     break;
                 };
 
+                let idle_inner_count = {
+                    let now = Instant::now();
+                    let Ok(mut scheduler_inners) = scheduler_pool.scheduler_inners.lock() else {
+                        break;
+                    };
+                    // I want to use the fancy ::extract_if() no matter what....
```

Comment:
why though? it seems like we can get something similar with retain and collecting a vec inside?

Reply:
> why though? it seems like we can get something similar with retain and collecting a vec inside?

`retain` only gives `&mut` to the predicate closure. So, we can't collect (i.e. take the ownership) in that way. After all, that's the why there's the still unstable `::extract_if()`.

And `::extract_if()` is a perfect match and shortest & fastest way for this use case.

Reply:
well, while waiting for lgtm, I've written a in-source commentary: https://github.com/anza-xyz/agave/pull/1575/commits/2b90fd1587ef87ba21154f8b4e5a14844962babd

Reply:
yeah agree it's a good match, would just love if the feature were stable. Just concerned with adding dependency on an otherwise unused crate. repo has no readme either.

kind of curious how you found it.
Looking at the repo, it seems fine to me as is - but who knows with upgrades.

Reply:
> kind of curious how you found it.

well, crates.io's own search functionality didn't work... i just googled...: https://www.google.com/search?q=rust+extract_if lol



> Looking at the repo, it seems fine to me as is - but who knows with upgrades.

as for upgrades, that's certainly a possibility. however, our dep is pinned to exact commit hash via `Cargo.lock` as usual. so, i think it's okay unless we're practicing blind dependabot merges. ;)

also as far as i screened author profile pages (https://github.com/chyyran and https://crates.io/users/chyyran). i think there's only legit activity.

Lastly, I'm planning to advertise this crate at the rust's tracking issue to increase the exposure of this crate across the ecosystem and to move this forward the stabilization.